src/mlang/optimizing_ir/partial_evaluation.ml

```diff
@@ -312,7 +311,7 @@ let rec partially_evaluate_expr (ctx : partial_ev_ctx) (p : Mir.program)
               from_literal
                 (Bir_interpreter.evaluate_expr p
                    (Pos.same_pos_as (Mir.Binop (op, new_e1, new_e2)) e1)
-                   RegularFloat)
+                   RegularFloat RODefault)
```

Comment:
Couldn't you get the right rounding mode here by adding a `Cli.rounding_mode = ref ...` global that is set by `driver.ml` ? Sounds like an easy fix.

Reply:
Yeah, I could to that.
How about doing that also for the number representation too ?

Reply:
Yep you can do that. Those things are not meant to change during compiler execution.

Reply:
Actually, that leads to a cycle between Cli and Bir_interpreter. That requires me to either :
- move the types `round_ops` and  `value_sort` out of Bir_interpreter
- move the global options out of Cli
- something else ?
What solution du you prefer ?

Reply:
I would be in favor of moving `round_ops` and `value_sorts` into `Cli` to break the cycle.